detectron2-semantic-segmentation/{{cookiecutter and 'code'}}/dataset.py

```diff
@@ -0,0 +1,149 @@
+import os
+import random
+import cv2
+import json
+import glob
+import numpy as np
+import pandas as pd
+import itertools
+import multiprocessing as mp
+import functools
+from detectron2.structures import BoxMode
+
+from util import get_class_names
+
+# adapted from https://www.kaggle.com/paulorzp/run-length-encode-and-decode
+def rle_decode(mask_rle, shape):
+    '''
+    mask_rle: run-length as string formated (start length)
+    shape: (height,width) of array to return
+    Returns numpy array, 1 - mask, 0 - background
+    '''
+    s = mask_rle.split()
+    starts, lengths = [np.asarray(x, dtype=int) for x in (s[0:][::2], s[1:][::2])]
+    starts -= 1
+    ends = starts + lengths
+    img = np.zeros(shape[0]*shape[1], dtype=np.uint8)
+    for lo, hi in zip(starts, ends):
+        img[lo:hi] = 1
+    return img.reshape(shape)
+
+# adapted from https://stackoverflow.com/questions/49494337/encode-numpy-array-using-uncompressed-rle-for-coco-dataset
+def binary_mask_to_rle(binary_mask):
+    rle = {'counts': [], 'size': list(binary_mask.shape)}
+    counts = rle.get('counts')
+    for i, (value, elements) in enumerate(itertools.groupby(binary_mask.ravel(order='F'))):
+        if i == 0 and value == 1:
+            counts.append(0)
+        counts.append(len(list(elements)))
+    return rle
+
+def split_data(file_list):
+    random.seed(0)
+    random.shuffle(file_list)
+
+    # split into train and validation sets
+    split = len(file_list)*90//100
+    train_list = file_list[:split]
+    val_list = file_list[split:]
+    return train_list, val_list
+
+def get_id(filename):
+    tokens = filename.split('/')
+    return tokens[4] + '_' + '_'.join(tokens[-1].split('_')[0:2])
```

Comment:
This appears to be incorrect. It works on Luminide but on Kaggle it is incorrect due to an extra layer in the directory structure. Adjusting `tokens[4]` to `tokens[-3]` should do the trick. Verifying right now.

```python3
def get_id(filename):
    tokens = filename.split('/')
    return tokens[-3] + '_' + '_'.join(tokens[-1].split('_')[0:2])
print(get_id('../input/uw-madison-gi-tract-image-segmentation/train/case101/case101_day20/scans/slice_0001_266_266_1.50_1.50.png'))
print(get_id('../input/train/case101/case101_day20/scans/slice_0001_266_266_1.50_1.50.png'))
```
```
case101_day20_slice_0001
case101_day20_slice_0001
```

Reply:
Confirmed via submitting with code that skips the prediction but uses all code except for the model/prediction. No error now.
<img width="972" alt="Screen Shot 2022-05-18 at 1 43 38 PM" src="https://user-images.githubusercontent.com/1946536/169152430-ba849afb-18da-45ad-a8b1-9b24ae0c1795.png">